File: Nível 1/3058 - Supermercado/main.lua
quantia = tonumber(io.read())

menor = 9999999999999

for i = 1, quantia do
    infos = io.read() 
    preco, gramas = infos:match("(%S+)%s+(%S+)")
    calc = (preco / gramas) * 1000

    if (calc < menor) then 
        menor = calc 
    end
end

print(string.format("%.2f", menor))

File: Nível 1/3058 - Supermercado/main.py
quantia = int(input("A: "))

menor = 9999999999999999999999

for i in range(1, quantia + 1):
    preco, gramas = map(float, input("B: ").split(" "))
    calc = (preco / gramas) * 1000

    if (calc < menor):
        menor = calc 

print(f"{menor:.2f}")